packages/jest-config/src/__tests__/normalize.test.ts

```diff
@@ -1918,3 +1919,20 @@ describe('updateSnapshot', () => {
     Defaults.ci = defaultCiConfig;
   });
 });
+
+describe('snapshotFormat', () => {
+  it('should accept custom `compareKeys`', async () => {
+    const compareKeys = () => 0;
+
+    const {options} = await normalize(
+      {
+        ci: false,
+        rootDir: '/root/',
+        snapshotFormat: {compareKeys} as SnapshotFormat,
```

Comment:
Those casts could be removed with such a patch:
```diff
diff --git a/packages/jest-config/src/ValidConfig.ts b/packages/jest-config/src/ValidConfig.ts
index db7ea5fe6d..1a4348d5dc 100644
--- a/packages/jest-config/src/ValidConfig.ts
+++ b/packages/jest-config/src/ValidConfig.ts
@@ -5,7 +5,6 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import type {SnapshotFormat} from '@jest/schemas';
 import type {Config} from '@jest/types';
 import {replacePathSepForRegex} from 'jest-regex-util';
 import {multipleValidOptions} from 'jest-validate';
@@ -113,8 +112,8 @@ const initialOptions: Config.InitialOptions = {
   slowTestThreshold: 5,
   snapshotFormat: {
     ...PRETTY_FORMAT_DEFAULTS,
-    compareKeys: () => {},
-  } as SnapshotFormat,
+    compareKeys: () => 0,
+  },
   snapshotResolver: '<rootDir>/snapshotResolver.js',
   snapshotSerializers: ['my-serializer-module'],
   testEnvironment: 'jest-environment-node',
diff --git a/packages/jest-config/src/__tests__/normalize.test.ts b/packages/jest-config/src/__tests__/normalize.test.ts
index fb229b4e33..7b858656ee 100644
--- a/packages/jest-config/src/__tests__/normalize.test.ts
+++ b/packages/jest-config/src/__tests__/normalize.test.ts
@@ -9,7 +9,6 @@
 import {createHash} from 'crypto';
 import path from 'path';
 import semver = require('semver');
-import type {SnapshotFormat} from '@jest/schemas';
 import type {Config} from '@jest/types';
 import {escapeStrForRegex} from 'jest-regex-util';
 import Defaults from '../Defaults';
@@ -1928,11 +1927,11 @@ describe('snapshotFormat', () => {
       {
         ci: false,
         rootDir: '/root/',
-        snapshotFormat: {compareKeys} as SnapshotFormat,
+        snapshotFormat: {compareKeys},
       },
       {} as Config.Argv,
     );
 
-    expect((options.snapshotFormat as any).compareKeys).toBe(compareKeys);
+    expect(options.snapshotFormat.compareKeys).toBe(compareKeys);
   });
 });
diff --git a/packages/jest-schemas/src/index.ts b/packages/jest-schemas/src/index.ts
index b3f0a32eeb..09a31cff5b 100644
--- a/packages/jest-schemas/src/index.ts
+++ b/packages/jest-schemas/src/index.ts
@@ -10,6 +10,9 @@ import {Static, Type} from '@sinclair/typebox';
 const RawSnapshotFormat = Type.Partial(
   Type.Object({
     callToJSON: Type.Readonly(Type.Boolean()),
+    compareKeys: Type.Readonly(
+      Type.Function([Type.String(), Type.String()], Type.Number()),
+    ),
     escapeRegex: Type.Readonly(Type.Boolean()),
     escapeString: Type.Readonly(Type.Boolean()),
     highlight: Type.Readonly(Type.Boolean()),
```
But I'm not sure if I should apply it because it feels like maybe `compareKeys` was omitted intentionally when declaring schemas in https://github.com/facebook/jest/pull/12384